framework/src/executioners/Transient.C

```diff
@@ -442,8 +469,50 @@ Transient::solveStep(Real input_dt)
   // Update warehouse active objects
   _problem.updateActiveObjects();
 
+  if (_prev_time == _time && _relax_factor != 1.0) // i.e. _pic_it > 0
+  {
+    NonlinearSystem & _nl_system = _fe_problem.getNonlinearSystem();
+    System & libmesh_nl_system = _nl_system.system();
+    NumericVector<Number> & solution = *libmesh_nl_system.solution;
+    NumericVector<Number> & relax_previous = libmesh_nl_system.get_vector("relax_previous");
+    solution.close();
+
+    // Save off the current solution
+    relax_previous = solution;
+    relax_previous.close();
+
+    // Snag all of the local dof indices for all of these variables
+    AllLocalDofIndicesThread aldit(libmesh_nl_system, _relaxed_vars);
+    ConstElemRange & elem_range = *_fe_problem.mesh().getActiveLocalElementRange();
+    Threads::parallel_reduce(elem_range, aldit);
+
+    _relaxed_dofs = aldit._all_dof_indices;
+  }
+
   _time_stepper->step();
 
+  if (_prev_time == _time && _relax_factor != 1.0) // i.e. _pic_it > 0
+  {
+    NonlinearSystem & _nl_system = _fe_problem.getNonlinearSystem();
+    System & libmesh_nl_system = _nl_system.system();
+    NumericVector<Number> & solution = *libmesh_nl_system.solution;
+    NumericVector<Number> & relax = libmesh_nl_system.get_vector("relax");
+    NumericVector<Number> & relax_previous = libmesh_nl_system.get_vector("relax_previous");
+    relax = solution;
+    solution.close();
+    relax.close();
+    relax_previous.close();
+    for (const auto & dof : _relaxed_dofs)
+    {
+      solution.set(dof,
+                   (relax_previous(dof) * (1.0 - _relax_factor)) + (relax(dof) * _relax_factor));
+    }
+    solution.close();
```

Comment:
I think you will need to call `libmesh_nl_system.update()` here because you changed the solution of the non-ghosted version, i.e. `solution` here, and you need to update the system so that the values will populate to the ghosted solution, `current_local_solution` on the system.